src/swm.rs

```diff
@@ -100,67 +174,13 @@ pub struct Parts {
 /// PMU.
 ///
 /// [module documentation]: index.html
-pub struct Handle<State = init_state::Enabled> {
+pub struct Handle {
     swm: pac::SWM0,
-    _state: State,
 }
 
-impl Handle<init_state::Enabled> {
```

Comment:
I think this should stay on `Handle`. The clock is only required for the switch matrix interface. The peripheral itself doesn't need a clock to function. Configuring the switch matrix, then disabling its clock to save power, is completely valid, but no longer possible with your changes.

Reply:
Ah, knew there had to be a reason for this